airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -83,4 +132,347 @@ protected DataType<?> toDialectType(final AirbyteProtocolType airbyteProtocolTyp
+    final List<Field<?>> fields =
+        metaColumns.entrySet().stream().map(metaColumn -> field(quotedName(metaColumn.getKey()), metaColumn.getValue())).collect(Collectors.toList());
+    final List<Field<?>> dataFields =
+        columns.entrySet().stream().map(column -> field(quotedName(column.getKey().name()), toDialectType(column.getValue()))).collect(
+            Collectors.toList());
+    dataFields.addAll(fields);
+    return dataFields;
+  }
+
+  /**
+   * Use this method to get the final table meta columns with or without _airbyte_meta column.
+   *
+   * @param includeMetaColumn
+   * @return
+   */
+  LinkedHashMap<String, DataType<?>> getFinalTableMetaColumns(boolean includeMetaColumn) {
+    final LinkedHashMap<String, DataType<?>> metaColumns = new LinkedHashMap<>();
+    metaColumns.put(COLUMN_NAME_AB_RAW_ID, SQLDataType.VARCHAR(36).nullable(false));
+    metaColumns.put(COLUMN_NAME_AB_EXTRACTED_AT, SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false));
+    if (includeMetaColumn)
+      metaColumns.put(COLUMN_NAME_AB_META, getStructType().nullable(false));
+    return metaColumns;
+  }
+
+  /**
+   * build jooq fields for raw table with type-casted data columns first and then meta columns without
+   * _airbyte_meta.
+   *
+   * @param columns
+   * @param metaColumns
+   * @return
+   */
+  @VisibleForTesting
+  List<Field<?>> buildRawTableSelectFields(final LinkedHashMap<ColumnId, AirbyteType> columns, final Map<String, DataType<?>> metaColumns) {
+    final List<Field<?>> fields =
+        metaColumns.entrySet().stream().map(metaColumn -> field(quotedName(metaColumn.getKey()), metaColumn.getValue())).collect(Collectors.toList());
+    // Use originalName with non-sanitized characters when extracting data from _airbyte_data
+    final List<Field<?>> dataFields = extractRawDataFields(columns);
+    dataFields.addAll(fields);
+    return dataFields;
+  }
+
+  @VisibleForTesting
+  Condition rawTableCondition(DestinationSyncMode syncMode, boolean isCdcDeletedAtPresent, Optional<Instant> minRawTimestamp) {
+    Condition condition = field(name(COLUMN_NAME_AB_LOADED_AT)).isNull();
+    if (syncMode == DestinationSyncMode.APPEND_DEDUP) {
+      if (isCdcDeletedAtPresent) {
+        condition = condition.or(cdcDeletedAtNotNullCondition());
+      }
+    }
+    if (minRawTimestamp.isPresent()) {
+      condition = condition.and(field(name(COLUMN_NAME_AB_EXTRACTED_AT)).gt(minRawTimestamp.get().toString()));
+    }
+    return condition;
+  }
+
+  @Override
+  public String createTable(final StreamConfig stream, final String suffix, final boolean force) {
+    // TODO: Use Naming transformer to sanitize these strings with redshift restrictions.
+    String finalTableIdentifier = stream.id().finalName() + suffix.toLowerCase();
+    if (!force) {
+      return Strings.join(
+          List.of(
+              createSchemaSql(stream.id().finalNamespace()),
```

Comment:
(nonblocking) this will need to be refactored after @jbfbell 's https://github.com/airbytehq/airbyte/pull/33124/, but should be a pretty easy change.